src/framework/theme/components/toggle/toggle.component.ts

```diff
@@ -0,0 +1,271 @@
+ *
+ * @styles
+ *
+ * toggle-height:
+ * toggle-width:
+ * toggle-border-width:
+ * toggle-border-radius:
+ * toggle-border-color:
+ * toggle-background-color:
+ * toggle-outline-width:
+ * toggle-outline-color:
+ * toggle-switcher-size:
+ * toggle-switcher-background-color:
+ * toggle-switcher-checkmark-color:
+ * toggle-disabled-background-color:
+ * toggle-disabled-border-color:
+ * toggle-disabled-checkmark-color:
+ * toggle-primary-background-color:
+ * toggle-primary-border-color:
+ * toggle-primary-checked-background-color:
+ * toggle-primary-checked-border-color:
+ * toggle-primary-switcher-checkmark-color:
+ * toggle-primary-focus-border-color:
+ * toggle-primary-hover-background-color:
+ * toggle-primary-hover-border-color:
+ * toggle-primary-active-background-color:
+ * toggle-primary-active-border-color:
+ * toggle-success-background-color:
+ * toggle-success-border-color:
+ * toggle-success-checked-background-color:
+ * toggle-success-checked-border-color:
+ * toggle-success-switcher-checkmark-color:
+ * toggle-success-focus-border-color:
+ * toggle-success-hover-background-color:
+ * toggle-success-hover-border-color:
+ * toggle-success-active-background-color:
+ * toggle-success-active-border-color:
+ * toggle-info-background-color:
+ * toggle-info-border-color:
+ * toggle-info-checked-background-color:
+ * toggle-info-checked-border-color:
+ * toggle-info-switcher-checkmark-color:
+ * toggle-info-focus-border-color:
+ * toggle-info-hover-background-color:
+ * toggle-info-hover-border-color:
+ * toggle-info-active-background-color
+ * toggle-info-active-border-color:
+ * toggle-warning-background-color:
+ * toggle-warning-border-color:
+ * toggle-warning-checked-background-color:
+ * toggle-warning-checked-border-color:
+ * toggle-warning-switcher-checkmark-color:
+ * toggle-warning-focus-border-color:
+ * toggle-warning-hover-background-color:
+ * toggle-warning-hover-border-color:
+ * toggle-warning-active-background-color:
+ * toggle-warning-active-border-color:
+ * toggle-danger-background-color:
+ * toggle-danger-border-color:
+ * toggle-danger-checked-background-color:
+ * toggle-danger-checked-border-color:
+ * toggle-danger-switcher-checkmark-color:
+ * toggle-danger-focus-border-color:
+ * toggle-danger-hover-background-color:
+ * toggle-danger-hover-border-color:
+ * toggle-danger-active-background-color:
+ * toggle-danger-active-border-color:
+ */
+@Component({
+  selector: 'nb-toggle',
+  animations: [
+    trigger('onOff', [
+      state('ltrOn', ltrState, defaultState),
+      state('rtlOn', rtlState, defaultState),
+      transition(':enter', [animate(0)]),
+      transition('* <=> *', [animate('0.25s')]),
+    ]),
+  ],
+  template: `
+    <label class="toggle-label">
+      <input type="checkbox" class="native-input visually-hidden"
+             [disabled]="disabled"
+             [checked]="checked"
+             (change)="updateValue($event)"
+             (blur)="setTouched()"
```

Comment:
Also please stop propagation of input click event as otherwise there will be two click events coming from toggle (one from label and one from input)